discovery/device/src/DiscoveryDeviceClient.cs

```diff
@@ -65,7 +65,7 @@ public class DiscoveryDeviceClient
         /// </summary>
         /// <param name="cancellationToken"></param>
         /// <returns>nonce for authentication challenge</returns>
-        public Task<string> IssueChallengeAsync(CancellationToken cancellationToken = default)
+        public Task<byte[]> IssueChallengeAsync(CancellationToken cancellationToken = default)
```

Comment:
I believe the contract is IssueChallengeAsync gives a nonce which is then later passed along the GetOnboardingInfoAsync call? Is that right?
If yes, do the public signature of these two methods need to change? Is there any interpretation of the nonce in the consumer of the SDK?

Reply:
I'm ok with the change from string to byte[] as such, but want to ensure that we are conscious about what changes results in the change in the consumer code (which the SDK layer is normally supposed to abstract out)

Reply:
> I believe the contract is IssueChallengeAsync gives a nonce which is then later passed along the GetOnboardingInfoAsync call? Is that right?

Yes that is correct

There is no client interpretation of the nonce but this change is cleaner overall since the alternative would be converting to string, then converting from string back to byte array within the SDK.

Reply:
Yes, I was ok with the change given that we haven't shipped anything yet... However these are areas to keep in mind once things go out in the field.